cmd/xl-v1-healing-common.go

```diff
@@ -133,3 +137,61 @@ func xlShouldHeal(partsMetadata []xlMetaV1, errs []error) bool {
 	}
 	return false
 }
+
+// xlHealStat - returns a structure which describes how many erasure
+// chunks are missing, how many bytes of the object needs to be healed
+// and if it is possible to heal with the chunks that are present.
+func xlHealStat(xl xlObjects, partsMetadata []xlMetaV1, errs []error) HealInfo {
+	missingDataCount := 0
+	missingParityCount := 0
+	quorum := xl.readQuorum
+	successCount := 0
+
+	// Compute heal statistics like bytes to be healed, missing
+	// data and missing parity count.
+	for i, err := range errs {
+		if err == nil {
+			successCount++
+			continue
+		}
+
+		// xl.json is not found, which implies the erasure
+		// coded chunks are unavailable in the corresponding disk.
+		// First half of the disks are data and the rest are parity.
```

Comment:
Actually this logic is not true. - data and parity are truly randomized so basically object1's data disk is perhaps a parity disk for object2 . This shifting of disks happens for each `object` name.